packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/V1_ProtocolToMetaModelPropertyMappingVisitor.ts

```diff
@@ -187,9 +187,16 @@ export class V1_ProtocolToMetaModelPropertyMappingVisitor
         );
       }
     }
-    const sourceSetImplementation: SetImplementation = protocol.source
-      ? topParent.parent.getClassMapping(protocol.source)
-      : topParent;
+    let sourceSetImplementation: SetImplementation = topParent;
+    if (protocol.source) {
+      try {
+        sourceSetImplementation = topParent.parent.getClassMapping(
+          protocol.source,
+        );
+      } catch (e: unknown) {
+        // ignore
+      }
+    }
```

Comment:
can we use `returnUndefOnError`, which does the try catch and returns undefined on error.
so if it returns undefined we use top parent

Reply:
good point!